cmd/internal/dgather.go

```diff
@@ -361,18 +363,50 @@ func RenderJobBPM(currentJob manifest.Job, jobInstances []manifest.JobInstance,
 			}
 
 			// Parse a rendered bpm.yml into the bpm Config struct
-			jobInstances[i].BPM, err = bpm.NewConfig(bpmBytes)
+			jobIndexBPM[i], err = bpm.NewConfig(bpmBytes)
 			if err != nil {
 				return err
 			}
+			instanceBytes := generateJobInstanceSHA(currentJob, jobInstances[i])
+
+			// All instances SHA are the same, while all instances are based on index 0
```

Comment:
I don't understand this comment, can you please explain why we need the SHA, and what do you mean that all instances are based on index 0?

Reply:
This is related to the previous comment, where I hardcoded the bpm rendering to use always the info of the job instance with index 0.  This will end with all job instances that are passed to bpm being the same, which make the fingerprint for each instance equal. This did not make a lot of sense to me, while to whole idea of sha´s is to be unique per instance.

Reply:
But the fingerprint is not used anywhere afaict.
Why keep it, for what purpose?

Reply:
This was based on Jan´s algorithm, because this is not longer needed, I just removed all logic related to the fingerprint generation.